lib/xhr.js

```diff
@@ -189,157 +179,169 @@ function enableXHROverH2(XMLHttpRequest, configuration) {
-                            self._changeState(XMLHttpRequest.LOADING, {'response': response});
-                        });
-                        response.on('finish', function () {
-                            if (configuration.debug) {
-                                configuration._log.debug("Got response for proxied XHR(" + destination.href + ")");
-                            }
-                            configuration.cache.put(requestInfo, response).then(function () {
-                                configuration._log.debug("Cache updated for proxied XHR(" + destination.href + ")");
-                            }, function (cacheError) {
-                                configuration._log.debug("Cache error for proxied XHR(" + destination.href + "):" + cacheError.message);                            
-                            }).finally(function () {
-                                self._changeState(XMLHttpRequest.DONE, {
-                                    'response': response
-                                });
+                        configuration.cache.put(requestInfo, response).then(function () {
+                            configuration._log.debug("Cache updated for proxied XHR(" + destination.href + ")");
+                        }, function (cacheError) {
+                            configuration._log.debug("Cache error for proxied XHR(" + destination.href + "):" + cacheError.message);                            
+                        }).finally(function () {
+                            self._changeState(XMLHttpRequest.DONE, {
+                                'response': response
                             });
                         });
                     });
+                });
 
-                    request.on('error', function (/*e*/) {
-                        // TODO, handle error
-                        // self._changeState('error');
-                    });
+                // TODO Why ?
+                request.on('error', function (/*e*/) {
+                    // TODO, handle error
+                    // self._changeState('error');
+                });
 
-                    // add to cache when receive pushRequest
-                    request.on('push', function(respo){
-                        configuration.onPush(respo);
-                    });
-                    // response.on('push', configuration.onPush);
+                // add to cache when receive pushRequest
+                request.on('push', function(respo){
+                    configuration.onPush(respo);
+                });
 
-                    if (body) {
-                        request.end(body);
-                    } else {
-                        request.end();
-                    }
-                    self.__dispatchEvent(new ProgressEvent('loadstart'));
-                }
+                // TODO why configuration.js line 240 ?
+                // response.on('push', configuration.onPush);
```

Comment:
This needs to be added back I think
` response.on('push', configuration.onPush);`

The issue is that push promises can come on any http2 stream.  Http2 does a new http2 stream for each request.  The on('push', latches on to push promises coming from that specific stream.

In most server implementations (like reaktivity) the push promise is delivered on any open stream.

The configuration.js opens a push stream via SSE in which push promises could be delivered.  But it is not guaranteed that push promises will only be delivered on that stream.  If we don't have this on, then we will miss any push promises that come on other request streams

Reply:
Oh nvm, please ignore, this can actually be removed.  I see it is on the response and not on the request.  The request is above